src/writer.rs

```diff
@@ -613,4 +658,49 @@ mod indentation {
 </outer>"#
         );
     }
+
+    #[cfg(feature = "serialize")]
+    #[derive(Serialize)]
+    struct Foo {
+        bar: Bar,
+        val: String,
+    }
+
+    #[cfg(feature = "serialize")]
+    #[derive(Serialize)]
+    struct Bar {
+        baz: usize,
+        bat: usize,
+    }
+
+    #[cfg(feature = "serialize")]
+    #[test]
+    fn element_writer_serialize() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+        let content = Foo {
+            bar: Bar { baz: 42, bat: 43 },
+            val: "foo".to_owned(),
+        };
+
+        writer
+            .create_element("paired")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_serializable_content(content)
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<paired attr1="value1" attr2="value2">
+    <Foo>
+        <bar>
+            <baz>42</baz>
+            <bat>43</bat>
+        </bar>
+        <val>foo</val>
+    </Foo>
+</paired>"#
+        );
```

Comment:
@Mingun  I would expect it to produce the same XML it would if you were to serialize it in a different context, but I don't think that would be the case if it produced the XML you suggest?

If you were to just serialize a `Foo`, then I would expect `<foo>...</foo>`

Reply:
Right, I asked myself this question and I can see both use cases being valid. Maybe we should leave this as an option for that function? Although I'm not sure how to tell the serializer to not produce the `<foo>` tags if we don't want them.

Reply:
Assuming it works as it does in your test case and in my description, it just feels extremely niche.  The `ElementWriter` construct only allows you to write one inner value wrapped in a pair of tags, and I can't think of many XML structures where you'd want 2 levels of nesting like that for only one possible value.  The construct is basically just too limiting to make it generally useful.

Wheras with the other approach of only supporting it on the writer, you can still accomplish the same thing, and it would be more flexible - if slightly more verbose.

```Rust
        writer
            .create_element("paired")
            .with_attribute(("attr1", "value1"))
            .with_attribute(("attr2", "value2"))
            .write_inner_content(|writer| {
                writer.write_serializable(&content1).unwrap();
                writer.write_serializable(&content2).unwrap();
                writer.write_serializable(&content3).unwrap();
            })?;
```

If it were to work the way @Mingun describes then none of this necessarily applies.  I'm just not sure I understand why the outer tags wouldn't be included when on the deserialization side you would expect some struct fields to potentially be sourced from attributes.  Shouldn't it work the same in the opposite direction?

Reply:
@Mingun So let's that `Foo` had some fields mapped to attributes, how does that work?  If the `Foo` element is erased and only its fields are serialized, any fields mapped to attributes will have nowhere to be serialized to.

I can appreciate that there are limitations with the serde model which may limit our expressiveness, but I would rather not do something than to do it in a way that is incorrect or illogical.

Reply:
```rust
#[derive(Serialize)]
struct Foo {
    #[serde(rename = "@attribute")]
    attribute: &'static str,

    element: Bar,
    list: Vec<&'static str>,

    #[serde(rename = "$text")]
    text: &'static str,
}
 let content = Foo {
    attribute: "attribute",
    element: Bar { baz: 42, bat: 43 },
    list: vec!["first element", "second element"],
    text: "text",
};

writer
    .create_element("paired")
    .with_attribute(("attr1", "value1"))
    .with_attribute(("attr2", "value2"))
    .write_serializable_content(content)
    .expect("failure");
```
I would expect (indentation depends on settings):
```xml
<paired attr1="value1" attr2="value2" attribute="attribute">
  <element>
    <baz>42</baz>
    <bat>43</bat>
  </element>
  <list>first element</list>
  <list>second element</list>
  text
</paired>
```